File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dict = {}
    for name, type in clothes:
        if type in dict.keys():
            # dict[type].append(name) # or dict[type] += [name]
            dict[type] += 1
        else:
            # dict[type] = [name]
            dict[type] = 1
    answer = 1
    for i in dict.values():
        answer *= i + 1
    return answer-1
            

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> closet = new HashMap<>();
        
        for (String[] c : clothes) 
            closet.put(c[1], closet.getOrDefault(c[1], 0) + 1);
        
        for (int count : closet.values())
            answer *= (count + 1); // 1은 안 입기
        
        return answer - 1;
    }
}